include/xgboost/tree_model.h

```diff
@@ -321,38 +299,28 @@ class TreeModel {
     fo->Write(&param, sizeof(TreeParam));
     CHECK_NE(param.num_nodes, 0);
     fo->Write(dmlc::BeginPtr(nodes_), sizeof(Node) * nodes_.size());
-    fo->Write(dmlc::BeginPtr(stats_), sizeof(NodeStat) * nodes_.size());
+    fo->Write(dmlc::BeginPtr(stats_), sizeof(RTreeNodeStat) * nodes_.size());
   }
   /*!
    * \brief add child nodes to node
    * \param nid node id to add children to
    */
   inline void AddChilds(int nid) {
-    int pleft  = this->AllocNode();
+    int pleft = this->AllocNode();
     int pright = this->AllocNode();
-    nodes_[nid].cleft_  = pleft;
-    nodes_[nid].cright_ = pright;
+    nodes_[nid].SetLeftChild(pleft);
+    nodes_[nid].SetRightChild(pright);
     nodes_[nodes_[nid].LeftChild() ].SetParent(nid, true);
     nodes_[nodes_[nid].RightChild()].SetParent(nid, false);
   }
-  /*!
-   * \brief only add a right child to a leaf node
-   * \param nid node id to add right child
-   */
-  inline void AddRightChild(int nid) {
-    int pright = this->AllocNode();
-    nodes_[nid].right  = pright;
-    nodes_[nodes_[nid].right].SetParent(nid, false);
-  }
   /*!
    * \brief get current depth
    * \param nid node id
-   * \param pass_rchild whether right child is not counted in depth
    */
-  inline int GetDepth(int nid, bool pass_rchild = false) const {
+  inline int GetDepth(int nid) const {
     int depth = 0;
     while (!nodes_[nid].IsRoot()) {
-      if (!pass_rchild || nodes_[nid].IsLeftChild()) ++depth;
```

Comment:
I don't quite understand the logic of this, so it's not used anymore?

Reply:
pass_rchild is never used in the code, therefore it is always false and this condition always evaluates to true. So I have removed both the function argument pass_rchild and the if statement so the behaviour is the same but the function is simpler.